### PR TITLE
Remove gfx906 from Jenkinsfile.  Most importantly, don't force gfx906 for vanilla codepath.

### PR DESCRIPTION
**Pull request by pcf000**
Forcing gfx906 for the vanilla codepath is what's causing the hangs on the public CI.

### CODE REVIEW

File: mlir/utils/jenkins/Jenkinsfile [hunk 1]
```diff
@@ -183,7 +183,7 @@ String getLabelFromCodepath(String codepath) {
         // and server nodes (gfx1030)
         label = 'mlir && ( gfx1030w || gfx1030 )'
     } else if (codepath == "vanilla"){
-        label = 'mlir && gfx906'
+        label = 'mlir'
```
Comment (krzysz00) [inline mlir/utils/jenkins/Jenkinsfile:186]: Given that both `gfx900` and `gfx906` don't exist in CI anymore, let's get rid of "vanilla" all together

Comment (pcf000) [reply mlir/utils/jenkins/Jenkinsfile:186]: Jerry wants to keep it.

Comment (krzysz00) [reply mlir/utils/jenkins/Jenkinsfile:186]: Why so? It's tested by navi2, no?

**Review by krzysz00 — APPROVED** [review]
Comment above aside, the broader removal of `vanilla` isn't a prerequisite for landing this, so it's approved

**Pull request merged**